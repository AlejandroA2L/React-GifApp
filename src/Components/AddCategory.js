import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) =>{
        setInputValue( e.target.value );
    }

    const handleSubmit = (e)=>{
e.preventDefault();//Obtiene el valor por defecto
if(inputValue.trim().length >2){
    setCategories(categ => [inputValue,...categ,]);//agregar el valor ingresado por el usr

}
    }
    //onChange evita que se haga refresh la pagina
    return (
        <form onSubmit={handleSubmit}>
           <input type="text" 
           value={inputValue} 
           onChange={handleInputChange}>
           </input>
           </form>
    )
 
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}


