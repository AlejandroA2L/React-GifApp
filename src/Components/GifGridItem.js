import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {

    //console.log(id,title,url);

    return (
        <div className="card animate__animated animate__fadeInDown">
            <img src={ url } alt={ title }></img>
            <p>{ title }</p>
        </div>
    )

    GifGridItem.propTypes={
        title: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
    }
}

/*
    1.Configurar Enzyme
    2.Enzyme to json
    3.Mostar el componente correctamente
        *shallow
        *wraper .toMatchSnapshot()
*/

