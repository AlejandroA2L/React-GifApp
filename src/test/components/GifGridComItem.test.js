import { GifGridItem } from "../../Components/GifGridItem"
import {shallow} from 'enzyme';
import React from 'react';
import { shallowToJson } from "enzyme-to-json"

describe('Pruebas en el GifGridItem', () => {
    
    const title = 'aja';
    const url = 'http://adsdasdasdsads.jpg'
    test('debe mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<GifGridItem title={title} url={url }></GifGridItem>)
        expect( wrapper ).toMatchSnapshot();
    })
    
})
