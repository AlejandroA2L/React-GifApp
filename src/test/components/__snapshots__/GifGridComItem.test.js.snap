// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="aja"
    src="http://adsdasdasdsads.jpg"
  />
  <p>
    aja
  </p>
</div>
`;
